crates/locutus-node/src/operations/subscribe.rs

```diff
@@ -268,27 +294,60 @@ where
             };
 
             if !op_storage.ring.has_contract(&key) {
-                //FIXME: should try forward to someone else who may have it first
-                // this node does not have the contract, return a void result to the requester
                 log::info!("Contract {} not found while processing info", key);
-                return Ok(return_err());
-            }
+                log::info!("Trying to found the contract from another node");
 
-            if let Err(_) = op_storage.ring.add_subscriber(key.clone(), subscriber) {
-                // max number of subscribers for this contract reached
-                return Ok(return_err());
-            }
+                let initial_htl = op_storage.ring.max_hops_to_live + 1;
+                let new_requester = target;
+                let new_target = op_storage.ring.closest_caching(&key, initial_htl, &[])[0];
+                let new_retries = retries + 1;
 
-            return_msg = state
-                .sm
-                .consume_to_output(SubscribeMsg::SeekNode {
-                    id,
-                    key,
-                    target,
-                    subscriber,
-                })?
-                .map(Message::from);
-            new_state = None;
+                // Retry seek node when the contract to subscribe has not been found in this node
+                conn_manager
+                    .send(
+                        new_target,
+                        (SubscribeMsg::SeekNode {
```

Comment:
After the return message, I don't think this would be passed back to the original requester but to the same node which then would do nothing with this message?

Reply:
i updated it in a newer commit